### DISCUSSION TITLE
Standard health check for JSON-RPC?

### DISCUSSION THREAD

**Discussion by unmaykr-aftermath** [paritytech/jsonrpsee#1494] [Q&A] [answered]
Hi, I'm new to the JSON-RPC standard. 

I found this library really useful to set up a microservice. However, I wonder if there's a standard way to do a 'health check' on the server without creating a special JSON-RPC method for it. Specifically, I want to set up a [HEALTHCHECK](https://docs.docker.com/reference/dockerfile/#healthcheck) in the `Dockerfile` for my JSON-RPC server. Any pointers?

**Comment by niklasad1** [comment, marked as answer]
Hey,

There is no "standard/embedded way" to do JSON-RPC health check but jsonrpsee provides a middleware to "proxy HTTP get request to a certain RPC handler" [example here](https://github.com/paritytech/jsonrpsee/blob/master/examples/examples/http_proxy_middleware.rs#L88-#L106) to help with that.

Thus, it really depends for your usecase if you want to handle it such as a separate endpoint on your server

**Reply by unmaykr-aftermath** [reply]
That's a great example. Thank you
